tiledb/sm/consolidator/fragment_meta_consolidator.cc

```diff
@@ -144,7 +144,14 @@ Status FragmentMetaConsolidator::consolidate(
   std::vector<Buffer> buffs(meta.size());
   auto status = parallel_for(
       storage_manager_->compute_tp(), 0, buffs.size(), [&](size_t i) {
-        RETURN_NOT_OK(meta[i]->write_footer(&buffs[i]));
+
+        SizeComputationSerializer size_computation_serializer;
+        meta[i]->write_footer(size_computation_serializer);
+        Tile tile{Tile::from_generic(size_computation_serializer.size())};
+        Serializer serializer(tile.data(), tile.size());
+        meta[i]->write_footer(serializer);
+
+        RETURN_NOT_OK(buffs[i].write(tile.data(), tile.size()));
```

Comment:
We could convert std::vector<Buffer> buffs(meta.size()); to std::vector<unique_ptr<Tile>>.

Reply:
Not seeing what you expect to gain by that, not currently changed.
(Seems like that adds construction of an additional unique_ptr wrapper, when the buffer will still have to be constructed, and since the item is a local variable, either way they're going away... plus additional deref of the 'ptr', tho in particular code path that's probably not much of a total expense.)

Reply:
I'm sorry, I didn't realize github somewhat formatted this and made the comment unclear 😅. Let me try again...

We could convert:
`std::vector<Buffer> buffs(meta.size());` 
to 
`std::vector<unique_ptr<Tile>>`

Reply:
Would `vector<Tile>` work instead of `vector<unique_ptr<Tile>>`?